packages/block-editor/src/components/rich-text/file-paste-handler.native.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlobURL } from '@wordpress/blob';
+
+export function filePasteHandler(images, html) {
```

Comment:
I'm trying to understand the differences with the web on this one, and what I'm seeing:

- The native version adds a 'wp-image-{uploadId}' class. This seems related to the media upload in the apps, but I'm not sure if it's actually used, since pasting an image on RichText would end up transforming to an Image block, and that would request a media import anyway.
- The web version receives a [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object, while native receives paths to the files. I imagine we'll see more of this distinction elsewhere and I wonder if we should override `@wordpress/blob` instead so that our "blob" URLs are actually the passed file URLs (i.e. `createBlobURL = url => url`).
- The web only processes files when there was no html pasted, but I think we could do the same.

Am I missing extra context, or could these two merge into a single implementation?

Reply:
> The native version adds a 'wp-image-{uploadId}' class. This seems related to the media upload in the apps, but I'm not sure if it's actually used, since pasting an image on RichText would end up transforming to an Image block, and that would request a media import anyway.

I think the reason the upload happens is because we set this wp-image-uploadId or else the image block will just refer to a local file. No?

Reply:
I'm not super familiar with the whole process, but what I see on the image block is that it will `requestMediaImport` for any image with a `file:` URL. https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/image/edit.native.js#L107

Reply:
> The web only processes files when there was no html pasted, but I think we could do the same.

Do you know the rational behind this? Why can't we have both? 

If we want to do this on the native side too, we will need to check for empty string because our paste code sends an empty HTML string when no HTML is available: https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift#L245

Reply:
It looks like it was a fix for one of those “fun” bugs https://github.com/WordPress/gutenberg/issues/4854

Reply:
> If we want to do this on the native side too, we will need to check for empty string

I think an empty string would evaluate as false, so the current check should work